src/tools/schema2graph.ts

```diff
@@ -102,7 +102,7 @@ export class SchemaNode {
   // schema name, if not possible the full schema address will be used. It will never the name
   // based off the internal counter (i.e. Internal$N pattern)
   humanName(connection: HandleConnectionSpec): string {
-    if (this.uniqueSchema || this.sources.length === 1) {
+    if ((!this.variableName && this.uniqueSchema) || this.sources.length === 1) {
```

Comment:
It's better to not rely on variable name not being an empty string, even though the parser does not allowed unnamed variables, we use them a lot in our tests and maybe someday we will allow them, e.g.

```
particle A
  data: reads ~ with {x: Text}
```

so I would prefer if you wrote `this.variableName === null`